pkg/modprovider/state.go

```diff
@@ -47,6 +47,8 @@ type moduleState struct {
 	rawState []byte
 	// Intended to store contents of TF lock file exactly.
 	rawLockFile []byte
+	// The map of module outputs gets passed explicitly with the state.
+	moduleOutputs resource.PropertyMap
```

Comment:
This is probably not needed anymore. Earlier versions tried to communicate the outputs on the wire, from module state Create/update back to the component resource, however due to secret floating that turned out to not be acceptable. So the latest version communicates the module outputs via the planStore side-channel.